ParameterTrendPercentages/ParameterTrendPercentages_1/ParameterTrendPercentages_1.cs

```diff
@@ -209,31 +208,92 @@ private void RetrieveTrendRecords()
         };
 
         var getTrendDataResponseMessage = (GetTrendDataResponseMessage)_callback.SendMessage(getTrendDataMessage);
-        List<TrendRecord> records = getTrendDataResponseMessage.Records.Values.FirstOrDefault().OrderBy(r => r.Time).ToList() ?? throw new DataMinerException("Failed to retrieve trend information for the provided timeslot");
+        var response = getTrendDataResponseMessage;
+
+        // Process trending information
+        Dictionary<string, double> result = new Dictionary<string, double>();
+        if (response.Records.Values.FirstOrDefault() != null)
+        {
+            List<TrendRecord> records = response.Records.Values.FirstOrDefault().OrderBy(r => r.Time).ToList();
+            Dictionary<DateTime, string> trendRecords = new Dictionary<DateTime, string>();
```

Comment:
After having looked through the code, a dictionary for this collection does not seem appropriate.